backend/routes.js

```diff
@@ -369,3 +369,63 @@ route.POST('/zkpp/updatePasswordHash/{contract}', {
 
   return Boom.internal('internal error')
 })
+
+// PWA push notification
+
+route.GET('/push/publickey', {}, function (req, h) {
+  return h.response(VAPID_PUBLIC_KEY).type('text/plain')
+})
+
+route.POST('/push/subscribe', {
+  validate: { payload: Joi.string().required() }
+}, function (req, h) {
+  const subscription = JSON.parse(req.payload)
+  // Reference: is it safe to use 'endpoint' as a unique identifier of a push subscription
+  // (https://stackoverflow.com/questions/63767889/is-it-safe-to-use-the-p256dh-or-endpoint-keys-values-of-the-push-notificatio)
+  pushSubscriptions.set(subscription.endpoint, subscription)
+
+  return { subscriptionId: subscription.endpoint }
+})
+
+route.DELETE('/push/unsubscribe/{subscriptionId}', {
+  validate: {
+    params: Joi.object({
+      subscriptionId: Joi.string().required()
+    })
+  }
+}, function (req, h) {
+  const { subscriptionId } = req.params
+
+  pushSubscriptions.delete(subscriptionId)
+  return { deletedId: subscriptionId }
+})
```

Comment:
Do we really need to create new routes for subscribing/unsubscribing when the websocket API already has these?